src/server/handlers/firestore/consents.ts

```diff
@@ -0,0 +1,225 @@
+  }
+}
+
+async function handleChallengeGeneration(server: Server, consent: Consent) {
+  if (!validator.isValidChallengeGeneration(consent)) {
+    throw new Error('Consent Object Missing Fields')
+  }
+
+  try {
+    server.app.mojaloopClient.postGenerateChallengeForConsent(
+      consent.consentId as string,
+      consent.party!.partyIdInfo.fspId as string
+    )
+  } catch (error) {
+    logger.error(error)
+  }
+}
+
+async function handleSignedChallenge(server: Server, consent: Consent) {
+  if (!validator.isValidSignedChallenge(consent)) {
+    throw new Error('Consent Object Missing Fields')
+  }
+
+  try {
+    server.app.mojaloopClient.putConsentId(
+      consent.consentId as string,
+      {
+        requestId: consent.id,
+        initiatorId: consent.initiatorId as string,
+        participantId: consent.participantId as string,
+        scopes: consent.scopes as TCredentialScope[],
+        credential: consent.credential as TCredential,
+      },
+      consent.party!.partyIdInfo.fspId as string
+    )
+  } catch (error) {
+    logger.error(error)
+  }
+}
+
+async function handleRevokingConsent(server: Server, consent: Consent) {
+  if (!validator.isValidRevokeConsent(consent)) {
+    throw new Error('Consent Object Missing Fields')
+  }
+
+  try {
+    // Make outgoing POST consents/{ID}/revoke request to Mojaloop
+    server.app.mojaloopClient.postRevokeConsent(
+      consent.consentId as string,
+      consent.party!.partyIdInfo.fspId as string
+    )
+  } catch (error) {
+    logger.error(error)
+  }
+}
+
+export const onCreate: ConsentHandler = async (
+  server: Server,
+  consent: Consent
+): Promise<void> => {
+  if (consent.status) {
+    // Skip transaction that has been processed previously.
+    // We need this because when the server starts for the first time,
+    // all existing documents in the Firebase will be treated as a new
+    // document.
+    return
+  }
+
+  await handleNewConsent(server, consent)
+}
+
+export const onUpdate: ConsentHandler = async (
+  server: Server,
+  consent: Consent
+): Promise<void> => {
+  if (!consent.status) {
+    // Status is expected to be null only when the document is created for the first
+    // time by the user.
+    logger.error('Invalid consent, undefined status.')
+    return
+  }
+
+  switch (consent.status) {
+    case ConsentStatus.PENDING_PARTY_LOOKUP:
+      await handlePartyLookup(server, consent)
+      break
+
+    case ConsentStatus.PENDING_PARTY_CONFIRMATION:
+      await handleConsentRequest(server, consent)
+      break
+
+    case ConsentStatus.AUTHENTICATION_REQUIRED:
+      await handleAuthentication(server, consent)
+      break
+
+    case ConsentStatus.CONSENT_GRANTED:
+      await handleChallengeGeneration(server, consent)
+      break
+
+    case ConsentStatus.ACTIVE:
+      await handleSignedChallenge(server, consent)
+      break
+
+    case ConsentStatus.REVOKE_REQUESTED:
+      await handleRevokingConsent(server, consent)
+      break
+  }
```

Comment:
lack of the default case would bring problems in the future, let define default and throw an exception if reached